src/android/SpectrumManager.java

```diff
@@ -39,98 +38,96 @@
-                            callbackContext.error("could not rename image");
-                            return;
-                        }
-                        PluginResult plugingResult = new PluginResult(PluginResult.Status.OK);
-                        plugingResult.setKeepCallback(true);
-                        callbackContext.sendPluginResult(plugingResult);
-                    } else {
-                        callbackContext.error("could not compress image");
+                    if (action.equals("compressImage")) {
+                        JSONObject config = (JSONObject) args.get(0);
+                        transcodeImage(config.getString("sourcePath"), config.getInt("maxSize"), callbackContext);
                     }
-
-                } catch (final IOException e) {
-                    callbackContext.error("source file does not exists");
-                } catch (final SpectrumException e) {
-                    callbackContext.error("invalid image");
+                } catch (Exception e) {
+                    callbackContext.error(e.getMessage());
+                    e.printStackTrace();
                 }
             }
         });
+        return true;
     }
 
+    private void transcodeImage(String path, int size, CallbackContext callbackContext) {
+        if (mSpectrum == null) {
+            SpectrumSoLoader.init(cordova.getActivity());
+            mSpectrum = Spectrum.make(new SpectrumLogcatLogger(Log.INFO), DefaultPlugins.get());
+        }
+        Uri tmpSrc = Uri.parse(path);
+        final Uri sourceUri = tmpSrc.getScheme() != null ? webView.getResourceApi().remapUri(tmpSrc) : tmpSrc;
+        final String sourcePath = sourceUri.toString();
+        File file = new File(sourcePath);
+        if (!file.exists()) {
+            callbackContext.error("source file does not exists");
+            return;
+        }
+        InputStream inputStream;
+        try {
+            inputStream = new FileInputStream(sourcePath);
+        } catch (Exception e) {
+            callbackContext.error(e.toString());
+            return;
+        }
+        final TranscodeOptions transcodeOptions;
+        ImageSize targetSize = getImageSize(path, size);
+        transcodeOptions = TranscodeOptions.Builder(new EncodeRequirement(JPEG, 80)).resize(ResizeRequirement.Mode.EXACT_OR_SMALLER, targetSize).build();
+        String fileExtension = MimeTypeMap.getFileExtensionFromUrl(Uri.fromFile(file).toString());
+        String destinationFileName = UUID.randomUUID().toString() + "_compressed." + fileExtension;
+        String destinationPath = sourcePath.replace(file.getName(), destinationFileName);
+        SpectrumResult result;
+        try {
+            result = mSpectrum.transcode(
+                    EncodedImageSource.from(inputStream),
+                    EncodedImageSink.from(destinationPath),
+                    transcodeOptions,
+                    "com.spectrum-plugin");
+        } catch (SpectrumException e) {
```

Comment:
```suggestion
        } catch (FileNotFoundException | SpectrumException e) {
```

Reply:
C'est pas possible d'ecrire cela en java

Reply:
https://www.javatpoint.com/java-7-catch-multiple-exceptions ?
https://stackoverflow.com/questions/3495926/can-i-catch-multiple-java-exceptions-in-the-same-catch-clause#3495968